### PR TITLE
fix: soft delete for sync run

### PR DESCRIPTION
**Pull request by afthabvp**
## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would impact existing functionality)
- [ ] Documentation update
- [ ] Chore
  
## Checklist:
- [x] Ensure a branch name is prefixed with `feature`, `bugfix`, `hotfix`, `release` or `chore` followed by `/` and branch name e.g `feature/add-salesforce-connector`
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/models/sync.rb [hunk 1]
```diff
@@ -47,9 +45,9 @@ class Sync < ApplicationRecord
 
   after_initialize :set_defaults, if: :new_record?
   after_save :schedule_sync, if: :schedule_sync?
-  after_discard :delete_workflow
+  after_discard :post_delete_sync
```
Comment (subintp) [inline app/models/sync.rb:48]: nit: handle_discard might be a better name. No need to have sync in the function name as it is already in the sync model

File: app/models/sync_run.rb [hunk 2]
```diff
@@ -77,6 +81,10 @@ def set_defaults
     self.failed_rows ||= 0
   end
 
+  def post_delete_sync_run
```
Comment (subintp) [inline app/models/sync_run.rb:84]: nit: handle_discard might be a better name. No need to have sync run in the function name as it is already in the sync model

**Review by subintp — APPROVED** [review]

File: app/models/sync_run.rb (hunk 2 above)
Comment (afthabvp) [reply app/models/sync_run.rb:84]: perform_post_discard_sync
is this okay?

File: app/models/sync.rb (hunk 1 above)
Comment (afthabvp) [reply app/models/sync.rb]: perform_post_discard_sync

**Review by subintp — APPROVED** [review]

**Pull request merged**